tcl/tools/tactics.tcl

```diff
@@ -51,20 +43,14 @@ namespace eval tactics {
         catch { set progressIncr [expr {602.0 / [llength $fileList]}] }
         busyCursor .
         foreach fname $fileList {
-            set name [file rootname [file nativename $fname]]
             set fname [file nativename $fname]
-            set baseId [sc_base slot $name]
-            if {$baseId == 0} {
-                progressBarSet $win.dummy 100 10
-                if { [catch { sc_base open [getBaseTypeFromFile $fname] $name } baseId] } {
-                    if {$::errorCode == $::ERROR::UserCancel} { break }
-                    ERROR::MessageBox
-                    continue
-                }
-                set wasOpened 0
-            } else  {
-                set wasOpened 1
+            lassign [::file::OpenOrSwitch $fname 0] err wasOpened
+            if { $err } {
+                if {$::errorCode == $::ERROR::UserCancel} { break }
```

Comment:
This does not seem to be correct.
I believe that file::Open_ will already have reported the error to the user.

Reply:
I have pushed a new branch (I have not tried the code) to illustrate how I think it should work:
https://github.com/benini/scid/commit/549362a3484840e424d30f270c626081c921ed1c

Please list and explain the changes to the API that are necessary to use it in tacticts.